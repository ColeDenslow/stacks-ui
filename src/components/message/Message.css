.stx-message {
  position: relative;
  padding: 1.25rem;
  border-left: 5px solid;
  margin-bottom: 1rem;
  border-color: #6B6B6B;
  background: #F8F8F8;
}

.stx-message .stx-message__dismiss-button {
  position: absolute;
  top: 1rem;
  right: 1rem;
  opacity: .5;
  margin-left: 0;
}

.stx-message--with-info {
  border-color: #0D5EFF;
  background: #F2F6FF; 
}

.stx-message--with-alert {
  border-color: #FF1A2D;
  background: #FFF3F4;
}

.stx-message--with-warning {
  border-color: #FAE719;
  background: #FEFDF3;
}

.stx-message--with-success {
  border-color: #69F224;
  background: #F7FEF3;
}
