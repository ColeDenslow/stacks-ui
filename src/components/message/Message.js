import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import Button from '../button/Button';

import './Message.css';

function Message(props) {
  const { className, children, dismissible, onDismiss, title, type, ...otherProps } = props;
  const isAlert = type === 'alert';
  const classNames = cn([
    'stx-message',
    { 'stx-message--with-alert': isAlert },
    { 'stx-message--with-warning': type === 'warning' },
    { 'stx-message--with-info': type === 'info' },
    { 'stx-message--with-success': type === 'success' },
    className,
  ]);

  return (
    <div
      className={classNames}
      role={isAlert ? 'alert' : 'status'}
      {...otherProps}
    >
      { title && <h2 className='stx-message__title'>{title}</h2> }

      { dismissible && <Button
        className='stx-message__dismiss-button'
        secondary
        onClick={onDismiss}>
        Dismiss
      </Button> }

      { children }
    </div>
  );
}

Message.defaultProps = {};

Message.propTypes = {
  /** Optional className for the component */
  className: PropTypes.string,

  /** Specify a type for the message to give it additional meaning */
  type: PropTypes.oneOf(['alert', 'warning', 'success', 'info', '']),

  /** A special heading for the message */
  title: PropTypes.string,

  /** Whether or not the message is dismissible */
  dismissible: PropTypes.bool,

  /** Event called when the dismiss button is clicked */
  onDismiss: PropTypes.func,

  /** Child elements */
  children: PropTypes.node,
};

export default Message;
