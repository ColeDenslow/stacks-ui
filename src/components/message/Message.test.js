import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Message from './Message';

describe('Message', () => {
  const onDismiss = () => 'onDismiss';

  test('Message renders with basic props', () => {
    const output = mount(<Message
      id='the-id'
      className='the-class'
    >
      Hello World
    </Message>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Message renders with title', () => {
    const output = mount(<Message
      id='the-id'
      className='the-class'
      title='title'
    >
      Hello World
    </Message>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Message renders with dismiss button', () => {
    const output = mount(<Message
      id='the-id'
      className='the-class'
      dismissible
      onDismiss={onDismiss}
    >
      Hello World
    </Message>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Message renders with type = alert', () => {
    const output = mount(<Message
      id='the-id'
      className='the-class'
      type='alert'
    >
      Hello World
    </Message>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Message renders with type = warning', () => {
    const output = mount(<Message
      id='the-id'
      className='the-class'
      type='warning'
    >
      Hello World
    </Message>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Message renders with type = info', () => {
    const output = mount(<Message
      id='the-id'
      className='the-class'
      type='info'
    >
      Hello World
    </Message>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Message renders with type = success', () => {
    const output = mount(<Message
      id='the-id'
      className='the-class'
      type='success'
    >
      Hello World
    </Message>);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('Message renders additional data props', () => {
    const output = mount(<Message
      id='the-id'
      className='the-class'
      data-additional-prop='additional-prop'
    >
      Hello World
    </Message>);

    expect(toJson(output)).toMatchSnapshot();
  });
});
