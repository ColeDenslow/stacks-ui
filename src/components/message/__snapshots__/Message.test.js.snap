// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Message renders additional data props 1`] = `
<Message
  className="the-class"
  data-additional-prop="additional-prop"
  id="the-id"
>
  <div
    className="stx-message the-class"
    data-additional-prop="additional-prop"
    id="the-id"
    role="status"
  >
    Hello World
  </div>
</Message>
`;

exports[`Message Message renders with basic props 1`] = `
<Message
  className="the-class"
  id="the-id"
>
  <div
    className="stx-message the-class"
    id="the-id"
    role="status"
  >
    Hello World
  </div>
</Message>
`;

exports[`Message Message renders with dismiss button 1`] = `
<Message
  className="the-class"
  dismissible={true}
  id="the-id"
  onDismiss={[Function]}
>
  <div
    className="stx-message the-class"
    id="the-id"
    role="status"
  >
    <Button
      block={false}
      className="stx-message__dismiss-button"
      disabled={false}
      href={null}
      loading={false}
      onClick={[Function]}
      primary={false}
      secondary={true}
      size={null}
      target={null}
      title={null}
      type={null}
    >
      <button
        aria-busy={false}
        aria-disabled={false}
        className="stx-button stx-button--secondary stx-message__dismiss-button"
        disabled={false}
        href={null}
        onClick={[Function]}
        target={null}
        title={null}
      >
        <span
          className="stx-button__text"
        >
          Dismiss
        </span>
        <LoadingSpinner
          className="stx-button__spinner"
        >
          <span
            className="stx-loading-spinner stx-button__spinner"
          />
        </LoadingSpinner>
      </button>
    </Button>
    Hello World
  </div>
</Message>
`;

exports[`Message Message renders with title 1`] = `
<Message
  className="the-class"
  id="the-id"
  title="title"
>
  <div
    className="stx-message the-class"
    id="the-id"
    role="status"
  >
    <h2
      className="stx-message__title"
    >
      title
    </h2>
    Hello World
  </div>
</Message>
`;

exports[`Message Message renders with type = alert 1`] = `
<Message
  className="the-class"
  id="the-id"
  type="alert"
>
  <div
    className="stx-message stx-message--with-alert the-class"
    id="the-id"
    role="alert"
  >
    Hello World
  </div>
</Message>
`;

exports[`Message Message renders with type = info 1`] = `
<Message
  className="the-class"
  id="the-id"
  type="info"
>
  <div
    className="stx-message stx-message--with-info the-class"
    id="the-id"
    role="status"
  >
    Hello World
  </div>
</Message>
`;

exports[`Message Message renders with type = success 1`] = `
<Message
  className="the-class"
  id="the-id"
  type="success"
>
  <div
    className="stx-message stx-message--with-success the-class"
    id="the-id"
    role="status"
  >
    Hello World
  </div>
</Message>
`;

exports[`Message Message renders with type = warning 1`] = `
<Message
  className="the-class"
  id="the-id"
  type="warning"
>
  <div
    className="stx-message stx-message--with-warning the-class"
    id="the-id"
    role="status"
  >
    Hello World
  </div>
</Message>
`;
